storage/s3.go

```diff
@@ -678,9 +688,29 @@ func newSession(opts Options) (*session.Session, error) {
 		sess.Config.Region = aws.String(endpoints.UsEast1RegionID)
 	}
 
+	// get region of the bucket and create session accordingly
+	// if the region is not provided, it means we want region-independent session
+	// for operations such as listing buckets, making a new bucket, ...
+	if opts.bucket != "" {
```

Comment:
URL creation fails if the bucket is empty (https://github.com/peak/s5cmd/blob/master/storage/url/url.go#L87-L89). Since we use bucket name in URL, do we still need this check for getting region info? Can we assume bucket is always a non-zero value?

Reply:
Since when getting a remote client, session for the bucket that the client will operate on is either created or fetched from cache, we face with issue when getting client for make-bucket operation as we would not be interacting with an existing bucket. Therefore, we have the [following change](https://github.com/peak/s5cmd/pull/194/files#diff-60ad7baa580083a1e939d08b56510758R71) which makes empty bucket receival possible.

Reply:
makes sense. thanks for the clarification 👍